### PR TITLE
Load md5 and Jcrop scripts and styles from their new locations.

### PR DESCRIPTION
**Pull request by ockham**
They could not be loaded as they have been replaced by their bower
counterparts with
owncloud/core@49fbd766c2ad6e5bc7c8401bfab2c20b1f442012 and
owncloud/core@73569b29bc3984e1e6df102faccb3c02c20573b3

### CODE REVIEW

**Comment by ockham** [comment]
Reviews, pls!

@Raydiation you mentioned this today around noon on `#owncloud-dev`
@jbtbnl @LukasReschke @MorrisJobke

**Comment by ockham** [comment]
CC'ing @babelouest @LEDfan and @wakeup

**Comment by jbtbnl** [comment]
@ockham is there a reason to switch to OC_Util? Is either one depreciated?

**Comment by ockham** [comment]
addVendorScript and addVendorStyle are only there, but not in \OCP\Util

**Comment by ockham** [comment]
alternative would be to use something like `\OCP\Util::addScript('vendor/blueimp-md5', 'md5');`, but addVendorScript sounded more to the point (it's what @MorrisJobke used in the commits I linked to)

**Comment by ockham** [comment]
Hm, technically, `OC_Util` is private. So maybe we should just add those two functions to the `OCP\Util`.

**Comment by jbtbnl** [comment]
OK, let's wait for a reply from one of the core developers then to hear what's the preferred method...

**Comment by ockham** [comment]
Okay, filed a PR: https://github.com/owncloud/core/pull/12039

**Comment by jbtbnl** [comment]
While I'm thinking about it, is it preferred to rely on libraries in core? It is clear that core developers are moving JS dependencies around, but it would be good to have a heads-up on devel@owncloud.org about this.

cc @MorrisJobke

**Comment by BernhardPosselt** [comment]
Check how the news app does it, you should load the scripts in the templates

**Comment by ockham** [comment]
@Raydiation Can I use the `style` and `script` commands for js and css files that are in core, too, or do I need to ship everything redundantly with the contacts app? (BT Are there any docs for those commands?)

**Comment by BernhardPosselt** [comment]
@ockham you should ship css and js files separately to prevent breakage like this in the future

**Comment by ockham** [comment]
Hm. I think there should really be some defined set of standard JS libraries along with their CSS files so apps don't have to duplicate them needlessly.

**Comment by BernhardPosselt** [comment]
Its a few kbs and youre in control of what you ship which is very convenient. It is actually the declared goal that your app is self contained, meaning that it has as little deps on core as needed

**Comment by MorrisJobke** [comment]
> @ockham is there a reason to switch to OC_Util? Is either one depreciated?

We're not finished with the transition and plan to move all the CSS and JS loading stuff from OC_Util to a more template related code path. Therefore we don't introduced the new public API because it will most likely removed in some days/weeks

**Comment by ockham** [comment]
Fair enough. I've installed bower and will use it to download those scripts into a `vendor`subdirectory.
Still, I would like to merge this PR until then, as I don't want to fix one issue with a PR that entails a lot of possible other issues (like using new versions etc.)

I've rolled back the second commit, so we don't depend on the availability of `addVendorStyle` and `addVendorScript` in `\OCP\Util` anymore, but can just apply it as-is.

**Comment by ockham** [comment]
The latest commit ea7e3b4 now moves loading of scripts and styles to the `contacts.php` template. Consider this an intermediate stage before shipping our own versions of the vendor scripts.

**Comment by BernhardPosselt** [comment]
Indeed, already looks pretty good, :+1: from my side

**Comment by jbtbnl** [comment]
@ockham yes, let's fix master for now :+1:

File: templates/contacts.php [hunk 1]
```diff
@@ -1,5 +1,43 @@
 <?php
 use OCA\Contacts\ImportManager;
+
+script('', [
```
Comment (MorrisJobke) [inline templates/contacts.php:4]: You could simply omit the empty string AFAIK

Comment (ockham) [inline templates/contacts.php:4]: I tried it - it doesn't work :-(

Comment (MorrisJobke) [inline templates/contacts.php:4]: Okay. Then I mixed this up :(

**Pull request merged**